protopipe/pipeline/event_preparer.py

```diff
@@ -739,9 +776,79 @@ def prepare_event(self, source, return_stub=True, save_images=False, debug=False
+                        bcolors.WARNING
+                        + "WARNING: This event was triggered with 1 LST image and <2 images from other telescope types."
+                        + "\nWARNING : direction reconstruction will not be performed."
+                        + bcolors.ENDC
+                    )
+
+                # Set all the involved images as NOT good for recosntruction
+                # even though they might have been
+                # but this is because of the LST stereo trigger....
+                for tel_id in event.r0.tel.keys():
+                    good_for_reco[tel_id] = 0
+                # and set the number of good and bad images accordingly
+                n_tels["GOOD images"] = 0
+                n_tels["BAD images"] = n_tels["Triggered"] - n_tels["GOOD images"]
+                # create a dummy container for direction reconstruction
+                reco_result = ReconstructedShowerContainer()
+
+                if return_stub:  # if saving all events (default)
+                    if debug:
+                        print(bcolors.OKBLUE + "Recording event..." + bcolors.ENDC)
+                        print(
+                            bcolors.WARNING
+                            + "WARNING: This is event shall NOT be used further along the pipeline."
+                            + bcolors.ENDC
+                        )
+                    yield stub(  # record event with dummy info
+                        event,
+                        mc_phe_image,
+                        dl1_phe_image,
+                        dl1_phe_image_mask_reco,
+                        dl1_phe_image_mask_clusters,
+                        good_for_reco,
+                        hillas_dict,
+                        hillas_dict_reco,
+                        n_tels,
+                        leakage_dict,
+                        concentration_dict
+                    )
+                    continue
+                else:
+                    continue
+
+            # Now in case the only triggered telescopes were
+            # - < self.min_ntel_LST LST,
+            # - >=2 any other telescope type,
+            # we remove the single-LST image and continue reconstruction with
+            # the images from the other telescope types
+            if self.LST_stereo and (n_triggered_LSTs < self.min_ntel_LST) and (n_triggered_LSTs != 0) and (n_triggered_non_LSTs >= 2):
+                if debug:
+                    print(
+                        bcolors.WARNING
+                        + f"WARNING: LST stereo trigger condition is active.\n"
+                        + f"This event triggered < {self.min_ntel_LST} LSTs "
+                        + f"and {n_triggered_non_LSTs} images from other telescope types.\n"
+                        + bcolors.ENDC
+                    )
+                for tel_id in triggered_LSTs:  # in case we test for min_ntel_LST>2
+                    if good_for_reco[tel_id]:
```

Comment:
This way you remove this explicit loop entirely

Reply:
good_for_reco is not a boolean mask, but a dictionary where the keys are tel_ids and the values are 1 if the cleaned+parametrized image passed all quality cuts, 0 otherwise.
It is then written to file

honestly, I would avoid trying to make this more fancier now considering that soon it will all disappear, unless it it clear that there is a bug due to how I wrote it

Reply:
so basically what is happening here is that I set the value to 0 only if the image passed the quality cuts (otherwise it is already been set to 0) thus I "remove" the single-LST image by not using it later for shower geometry reconstruction, but since I need to process it up to DL1b for the benchmarking I still need to know (via my dictionary of labels "good_for_reco") if it would have been used in other circumstances

Reply:
ok, if this is not meant to be long-term code, fine. This was just so that you could write out the used telescope pattern as we do in ctapipe, but maybe that is not needed since we will soon move to ctapipe's version